src/amo/components/InstallWarning/index.js

```diff
@@ -0,0 +1,184 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import type { AppState } from 'amo/store';
+import { ADDON_TYPE_EXTENSION, UNINSTALLED, UNKNOWN } from 'core/constants';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import tracking from 'core/tracking';
+import { withExperiment } from 'core/withExperiment';
+import Notice from 'ui/components/Notice';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { InstalledAddon } from 'core/reducers/installations';
+import type { WithExperimentInjectedProps } from 'core/withExperiment';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+|};
+
+type CouldShowWarningParams = {|
+  addonIsExtension: boolean,
+  addonIsRecommended: boolean | void,
+  experimentEnabled: $PropertyType<
+    WithExperimentInjectedProps,
+    'experimentEnabled',
+  >,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
```

Comment:
The `installStatus` is not `undefined` only when there is `mozAddonManager` IIRC, so in local dev, `installStatus` will always be `undefined` but it might also be `undefined` is a user disables `mozAddonManager`. The install button will work with or without `mozAddonManager` so I guess we need to take this use case into consideration.

I patched the code like this:

```diff
diff --git a/src/amo/components/InstallWarning/index.js b/src/amo/components/InstallWarning/index.js
index 0d2effb39..de84a2cc4 100644
--- a/src/amo/components/InstallWarning/index.js
+++ b/src/amo/components/InstallWarning/index.js
@@ -41,7 +41,8 @@ export const couldShowWarning = ({
     addonIsExtension &&
     !addonIsRecommended &&
     experimentEnabled &&
-    [UNINSTALLED, UNKNOWN].includes(installStatus)
+    ([UNINSTALLED, UNKNOWN].includes(installStatus) ||
+      installStatus === undefined)
   );
 };
```

Reply:
This is tricky, and that patch will work, but will cause a visual problem when `mozAddonManager` is installed, because `installStatus` can start with a value of `undefined` and then can update to its actual value e.g., `INSTALLED` at which point the UI will rerender, so the message flashes on the screen temporarily and then disappears in that case.

I discussed the whole `mozAddonManager` issue with @muffinresearch (I think it was on IRC as opposed to in an issue) and he agreed that we can just ignore the use case of when a user does not have `mozAddonManager` in production because that case is very rare.

If we absolutely want to deal with it I believe there's a way to check for the existence of `mozAddonManager`, and we could add that into the logic, but I think that's unnecessary.

Do you disagree @willdurand?

Reply:
> If we absolutely want to deal with it I believe there's a way to check for the existence of mozAddonManager, and we could add that into the logic, but I think that's unnecessary.

I think we need something like this. If we ever want to ignore the "no `mozAddonManager`" case, then we should not make it part of the experiment at all I believe.

Reply:
If `mozAddonManager` does not exist then `installStatus` will always be `undefined` (as you say above), and therefore, if we include `undefined` in the allowed statuses then we'd *always* show the warning, even when the add-on is currently installed. It seems like we need to choose to either always show the warning, or never show the warning, and I guess we chose the latter. Do you feel we should change that decision and always show the warning? It doesn't seem like there is a way to not choose one of those two paths, neither of which we'd really prefer.

Reply:
ah, that's correct.

I think we should always show the warning given that we don't know the status of the add-on *but* we do want to warn users (that's the goal, no?). So I think it'd be better to warn a user twice rather than not.